File: WeAre.Athenaeum.React.Toolkit/src/models/GeoLocation.ts
import GeoCoordinate from "./GeoCoordinate";

export default class GeoLocation extends GeoCoordinate {
    
    public country: string = "";

    public address: string = "";

    public city: string = "";

    public postalCode: string = "";

    public postalBox: string = "";

    public formattedAddress: string = "";

    public isGeoLocation: boolean = true;
}

File: Renta.Apps.Common/Models/Geo/GeoLocation.cs
using Renta.Apps.Common.Helpers;
using Renta.Apps.Common.Interfaces.Geo;

namespace Renta.Apps.Common.Models.Geo
{
    public sealed class GeoLocation : GeoCoordinate, IGeoLocation
    {
        public string Country { get; set; }

        public string Address { get; set; }

        public string City { get; set; }

        public string PostalCode { get; set; }

        public string PostalBox { get; set; }

        public string FormattedAddress => this.GetFormattedAddress(true);

        public int HashCode => FormattedAddress.ToLowerInvariant().GetHashCode();

        public bool IsGeoLocation => true;
    }
}